drivers/ethernet/intel/eth_intel_igc.c

```diff
@@ -0,0 +1,1363 @@
+/*
+ * Copyright (c) 2025 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eth_intel_igc, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <errno.h>
+#include <soc.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/crc.h>
+#include "../eth.h"
+#include "eth_intel_igc_priv.h"
+
+#define DT_DRV_COMPAT intel_igc_mac
+
+static uint32_t igc_read(mm_reg_t reg_addr, uint32_t offset)
+{
+	return sys_read32(reg_addr + offset);
+}
+
+static void igc_write(mm_reg_t reg_addr, uint32_t val)
+{
+	sys_write32(val, reg_addr);
+}
```

Comment:
these functions don't seem to to provide any real extra value, instead of directly using `sys_read32` and `sys_write32`. Why not using them directly?